app/services/dttp/register_for_trn.rb

```diff
@@ -22,15 +22,20 @@ def call
       batch_response = batch_request.submit
 
       entity_ids = OdataParser.entity_ids(batch_response: batch_response)
+      contact_dttp_id = entity_ids["contacts"].first[:entity_id]
 
-      trainee.trn_requested!(entity_ids["contacts"].first[:entity_id],
-                             entity_ids["dfe_placementassignments"].first[:entity_id])
+      trainee.trn_requested!(contact_dttp_id, entity_ids["dfe_placementassignments"].first[:entity_id])
 
       degree_change_set_ids.each do |degree, content_id|
         result = entity_ids["dfe_degreequalifications"].find { |item| item[:content_id] == content_id }
         degree.update!(dttp_id: result[:entity_id])
       end
 
+      # Has to be done after the contact entity is created - can't be done in the batch request
+      Dttp::UpdateTraineeStatus.call(status: DttpStatuses::PROSPECTIVE_TRAINEE_TRN_REQUESTED,
```

Comment:
We should put this in its own worker and queue it. If we leave it in here and it fails it will cause the batch above to be run again when the worker retries.